plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestingKuduServer.java

```diff
@@ -18,45 +18,76 @@
 import com.google.common.net.HostAndPort;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
 
 import java.io.Closeable;
 import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.UnknownHostException;
 import java.util.List;
 
+import static java.lang.String.format;
+
 public class TestingKuduServer
         implements Closeable
 {
+    private static final String KUDU_IMAGE = "apache/kudu:1.10.0";
     private static final Integer KUDU_MASTER_PORT = 7051;
     private static final Integer KUDU_TSERVER_PORT = 7050;
     private static final Integer NUMBER_OF_REPLICA = 3;
 
+    private static final String TOXIPROXY_IMAGE = "shopify/toxiproxy:2.1.0";
+    private static final String TOXIPROXY_NETWORK_ALIAS = "toxiproxy";
+
+    private final Network network;
+    private final ToxiproxyContainer toxiProxy;
     private final GenericContainer<?> master;
     private final List<GenericContainer<?>> tServers;
 
+    /**
+     * WORKAROUND WARNING:
+     * Kudu tablets needs to know the host/mapped port it will be bound to in order to configure --rpc_advertised_addresses
+     * However when using non-fixed ports in testcontainers, we only know the mapped port after the container starts up
+     * In order to workaround this, create a proxy to forward traffic from the host to the underlying tablets
+     * Since the ToxiProxy container starts up *before* kudu, we know the mapped port when configuring the kudu tablets
+     */
     public TestingKuduServer()
     {
-        Network network = Network.newNetwork();
+        network = Network.newNetwork();
         ImmutableList.Builder<GenericContainer<?>> tServersBuilder = ImmutableList.builder();
-        this.master = new GenericContainer<>("apache/kudu:1.10.0")
+
+        String hostIP = getHostIPAddress();
+
+        String masterContainerAlias = "kudu-master";
+        this.master = new GenericContainer<>(KUDU_IMAGE)
                 .withExposedPorts(KUDU_MASTER_PORT)
                 .withCommand("master")
+                .withEnv("MASTER_ARGS", format("--fs_wal_dir=/var/lib/kudu/master --logtostderr --use_hybrid_clock=false --rpc_bind_addresses=%s:%s", masterContainerAlias, KUDU_MASTER_PORT))
```

Comment:
Do we need this change for master server ?

Reply:
When debugging things, having the logs in stderr/available through docker logs was very helpful. 

The rpc bind address config is not strictly necessary since it binds to `0.0.0.0:7051` by default, so I can remove that from the master config.